packages/bundler-plugin-core/src/utils/getPreSignedURL.ts

```diff
@@ -25,14 +29,39 @@ export const getPreSignedURL = async ({
   uploadToken,
   serviceParams,
   retryCount = DEFAULT_RETRY_COUNT,
+  gitService,
 }: GetPreSignedURLArgs) => {
-  if (!uploadToken) {
-    red("No upload token found");
-    throw new NoUploadTokenError("No upload token found");
-  }
-
   const url = `${apiURL}/upload/bundle_analysis/v1`;
 
+  const headers = new Headers({
+    "Content-Type": "application/json",
+  });
+
+  const requestBody: RequestBody = serviceParams;
+  /**
+   * We currently require the branch to be in the format `owner:branch` to identify that it is a
+   * proper tokenless upload.
+   * See: https://github.com/codecov/codecov-api/pull/741
+   */
+  if (!uploadToken && serviceParams.branch?.includes(":")) {
+    if (gitService && gitService !== "") {
```

Comment:
```suggestion
    if (gitService) {
```

empty string will coerce to `false` so we don't need the additional check - ditto to wherever else you are doing this.